kroxylicious/src/main/java/io/kroxylicious/proxy/config/VirtualCluster.java

```diff
@@ -5,22 +5,73 @@
  */
 package io.kroxylicious.proxy.config;
 
-import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Optional;
 
 public class VirtualCluster {
 
     private final TargetCluster targetCluster;
+    private final ClusterEndpointProviderDefinition clusterEndpointProvider;
 
-    public VirtualCluster(@JsonProperty(value = "targetCluster") TargetCluster targetCluster) {
+    private final Optional<String> keyStoreFile;
```

Comment:
I think eventually there will be a tls node on the virtual cluster that will carry the cluster's certificate.